JustSaying.IntegrationTests/WhenRegisteringHandlersViaResolver/HandlerWithMessageContext.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+using System.Threading.Tasks;
+using JustSaying.IntegrationTests.TestHandlers;
+using JustSaying.Messaging.MessageHandling;
+
+namespace JustSaying.IntegrationTests.WhenRegisteringHandlersViaResolver
+{
+    class HandlerWithMessageContext : IHandlerAsync<OrderPlaced>
+    {
+        private readonly IMessageContextReader _contextReader;
+
+        public HandlerWithMessageContext(
+            Future<OrderPlaced> future,
+            IMessageContextReader contextReader)
+        {
+            Future = future;
+            _contextReader = contextReader;
```

Comment:
I was thinking the pattern we'd encourage would be to grab the context in the constructor. You get the same result either way, it keeps the code in the handler method a little bit tidier.
```csharp
_context = contextReader.MessageContext;
```